### PR TITLE
Tag-based release process

### LINKED ISSUE #4522: Make the release process non-blocking
See discussion on modeling-app channel.

1. Create a release issue on modeling-app to concentrate discussions on changelog and testing
2. Create a tag
3. Workflow starts on new tag, builds release artifacts and uploads them to a draft release
4. Grab the builds and follow the checklist, post test reports on the issue
5. Publish the release
6. Workflow starts on new published release, and pushes the builds to the public bucket and annouces it

### PR DESCRIPTION
**Pull request by pierremtb**
Needs #4514 to go in first ✅ 

Fixes #4522

**TL;DR: changes the main trigger for the release process from a Cut Release PR to a git tag, which likely should be on a commit to `main`, but not necessarily. The `build-apps` workflow builds the apps and creates a draft release at the end that we can edit content wise, and the `publish-apps-release` workflow takes care of the uploads to the public bucket once the release is published.**

Still supports fully automated nightly builds (no tags), and updater-test artifacts.

See https://github.com/KittyCAD/modeling-app/pull/4530/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5 for the updated process.

Manual test coverage of these updated YAML files:
- `build-apps` tested in nightly mode at https://github.com/KittyCAD/modeling-app/actions/runs/11960331136
- `build-apps` tested in release mode at https://github.com/KittyCAD/modeling-app/actions/runs/11959993043
- `publish-apps-release` tested at https://github.com/KittyCAD/modeling-app/actions/runs/11960866259/job/33345830067

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "zoo-modeling-app",
-  "version": "0.27.0",
+  "version": "0.0.0",
```
Comment (pierremtb) [inline package.json:3]: This is on purpose: this field is only changed during nightly or release builds. The version isn't tracked in the codebase anymore with this new process.

Comment (pierremtb) [reply package.json:3]: Ha that's a pretty good idea! Thanks. Instead of simply removing the `make-release.sh` script like now I guess we can just repurpose it to this :) Do you mind if I keep the name then?

Comment (nadr0) [reply package.json:3]: I'd probably give it a new name like `generate-semantic-release-tag.sh` because `make-release.sh` is not that descriptive and it isn't "making" anything.

Also I'd take my new source code above since I have edited the source of the previous tag and I edited the output so it only `echo` the tag. This means you can directly use the output of the bash script instead of copying and pasting the value from the terminal.

Comment (pierremtb) [reply package.json:3]: Ok cool that works, was thinking that this script could also create the tag and push it but we can for sure keep that manual

**Review by nadr0 — APPROVED** [review]
Thanks for making that addition to the semantic release process.

File: .github/workflows/build-apps.yml [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
   # Daily at 04:00 AM UTC
   # Will checkout the last commit from the default branch (main as of 2023-10-04)
 
 env:
-  CUT_RELEASE_PR: ${{ github.event_name == 'pull_request' && (contains(github.event.pull_request.title, 'Cut release v')) }}
-  NIGHTLY: ${{ github.event_name == 'schedule' }}
-  BUILD_RELEASE: ${{ github.event_name == 'release' || github.event_name == 'schedule' || github.event_name == 'pull_request' && (contains(github.event.pull_request.title, 'Cut release v')) }}
-  NOTES: ${{ github.event_name == 'release' && github.event.release.body || format('Non-release build, commit {0}', github.sha) }}
+  IS_RELEASE: ${{ github.ref_type == 'tag' }}
+  IS_NIGHTLY: ${{ github.event_name == 'schedule' }}
```
Comment (franknoirot) [inline .github/workflows/build-apps.yml:17]: [comment] Nice and easy to read

File: .github/workflows/build-apps.yml [hunk 3]
```diff
@@ -48,16 +47,17 @@ jobs:
...
+        if: ${{ env.IS_NIGHTLY == 'true' }}
         run: yarn files:flip-to-nightly
 
-      - name: Generate release notes
+      - name: Set release version
+        if: ${{ env.IS_RELEASE == 'true' }}
         run: |
-          echo "$NOTES" > release-notes.md
-          cat release-notes.md
+          export VERSION=${GITHUB_REF_NAME#v}
```
Comment (franknoirot) [inline .github/workflows/build-apps.yml:57]: [question] what does `#v` signify or do?

File: .github/workflows/build-apps.yml [hunk 4]
```diff
@@ -335,78 +351,44 @@ jobs:
...
-      - name: Upload release files to Github
-        if: ${{ github.event_name == 'release' }}
+      - name: Create draft release
         uses: softprops/action-gh-release@v2
+        if: ${{ env.IS_RELEASE == 'true' }}
         with:
+          name: ${{ env.VERSION }}
+          tag_name: ${{ env.VERSION }}
+          draft: true
+          generate_release_notes: true
```
Comment (franknoirot) [inline .github/workflows/build-apps.yml:393]: [comment] This eliminates the need to copy the release notes out of the terminal? Sick!

File: scripts/flip-files-to-nightly.sh [hunk 5]
```diff
@@ -1,13 +1,19 @@
...
-VERSION=$(date +'%-y.%-m.%-d') yarn files:set-version
+yarn files:set-version
 echo "$(jq --arg name 'Zoo Modeling App (Nightly)' '.productName=$name' package.json --indent 2)" > package.json
 
 # electron-builder.yml
 yq -i '.publish[0].url = "https://dl.zoo.dev/releases/modeling-app/nightly"' electron-builder.yml
 yq -i '.appId = "dev.zoo.modeling-app-nightly"' electron-builder.yml
 
+# Release notes
+echo "Nightly build $VERSION (commit $COMMIT)" > release-notes.md
```
Comment (franknoirot) [inline scripts/flip-files-to-nightly.sh:15]: [question] Oh cool so does the nightly build get automated release notes in the updater modal then?

**Review by franknoirot — APPROVED** [review]
LGTM just asked questions for my understanding, this is great

File: .github/workflows/build-apps.yml (hunk 3 above)
Comment (pierremtb) [reply .github/workflows/build-apps.yml:57]: @franknoirot it chops off the leading 'v'

File: .github/workflows/build-apps.yml (hunk 4 above)
Comment (pierremtb) [reply .github/workflows/build-apps.yml:393]: Yes! Same as checking the box when creating a release in the github UI

File: scripts/flip-files-to-nightly.sh (hunk 5 above)
Comment (pierremtb) [reply scripts/flip-files-to-nightly.sh:15]: No the release notes will just be this message, pointing to the version (which is a date YY.MM.DD) and a commit hash
